packages/shared/src/components/profile/UpvoterList.tsx

```diff
@@ -1,48 +1,73 @@
-import React, { ReactElement, memo, ReactNode } from 'react';
+import React, { ReactElement } from 'react';
 import Link from 'next/link';
-import { Upvote } from '../../graphql/common';
+import { UseInfiniteQueryResult } from 'react-query';
+import { Connection, HasConnection, Upvote } from '../../graphql/common';
 import { LazyImage } from '../LazyImage';
+import useFeedInfiniteScroll from '../../hooks/feed/useFeedInfiniteScroll';
 
-export interface UpvoterListProps {
-  upvotes: Upvote[];
-  children?: ReactNode;
+export interface UpvoterListProps<
+  TConnection extends HasConnection<Upvote, TKey>,
+  TKey extends keyof Connection<Upvote>,
+> {
+  queryResult: UseInfiniteQueryResult<TConnection>;
+  objectKey: TKey;
 }
 
-export function UpvoterList({
-  upvotes,
-  children,
-}: UpvoterListProps): ReactElement {
+export function UpvoterList<
+  TConnection extends HasConnection<Upvote, TKey>,
+  TKey extends keyof Connection<Upvote>,
+>({
+  objectKey,
+  queryResult,
+}: UpvoterListProps<TConnection, TKey>): ReactElement {
+  const canFetchMore =
+    !queryResult.isLoading &&
+    !queryResult.isFetchingNextPage &&
+    queryResult.hasNextPage &&
+    queryResult.data.pages.length > 0;
+
+  const infiniteScrollRef = useFeedInfiniteScroll({
+    fetchPage: queryResult.fetchNextPage,
+    canFetchMore,
+    enableTrackEngagement: false,
+  });
+
   return (
     <div className="flex flex-col relative">
-      {upvotes.map(({ user }) => (
-        <Link
-          key={user.username}
-          href={`https://app.daily.dev/${user.username}`}
-        >
-          <a className="flex flex-row hover:bg-theme-active px-6 py-3">
-            <LazyImage
-              imgSrc={user.image}
-              imgAlt={user.username}
-              className="w-12 h-12 rounded-10"
-              ratio="1 / 1"
-            />
-            <div className="flex flex-col flex-1 ml-4">
-              <span className="typo-callout font-bold">{user.name}</span>
-              <span className="typo-callout text-theme-label-secondary">
-                @{user.username}
-              </span>
-              {user.bio && (
-                <span className="mt-1 typo-callout text-salt-90">
-                  {user.bio}
+      {queryResult.data.pages.map((page) =>
+        page[objectKey].edges.map(({ node: { user } }) => (
+          <Link
+            key={user.username}
+            href={`https://app.daily.dev/${user.username}`}
```

Comment:
I think you can get the full URL from the GraphQL request. The field is called `permalink`